src/Mvc/Mvc.Core/src/Infrastructure/SystemTextJsonResultExecutor.cs

```diff
@@ -77,7 +74,11 @@ public async Task ExecuteAsync(ActionContext context, JsonResult result)
             var responseStream = response.Body;
             if (resolvedContentTypeEncoding.CodePage == Encoding.UTF8.CodePage)
             {
-                await JsonSerializer.SerializeAsync(responseStream, value, objectType, jsonSerializerOptions);
+                try
+                {
+                    await JsonSerializer.SerializeAsync(responseStream, value, objectType, jsonSerializerOptions, context.HttpContext.RequestAborted);
+                }
+                catch (OperationCanceledException) { }
                 await responseStream.FlushAsync();
```

Comment:
Should we include the cts here and run it inside the try-catch block?

Reply:
There's a variant of this where it gets transcoded. Should we also include the cts in that case?